src/peak_finding/1d_list.clj

```diff
@@ -0,0 +1,82 @@
+;; A peak is any element that is larger or equal to both it's neighbours
+;; Items that only have one neighbour (end items) are considered a peak if
+;; they are larger or equal to their neighbour
+;; This algorithm returns the first peak it finds.
+
+(ns peakfinder)
+
+(defn ^:dynamic find-peak
+  "multiple arity recursive function, for finding a peak in a list"
+  ([list] (->> (middle-val list) (item list) (find-peak list)))
+
+  ([list current-item]
+   (if (empty? list) -1
+       (let [index (:index current-item)
+             left (left-neighbour list current-item)
+             right (right-neighbour list current-item)]
+         (cond
+           (-> list (count) (- 1) (= index)) ;; Furthest item right
+           (peak? current-item left)
+
+           (= index 0) ;; Furthest item left
+           (peak? current-item right)
+
+           :else
+           (if (= (peak? current-item left right) -1)
+
+             (if (> (:value left) (:value current-item))
+               (find-peak list left) ;; Recursive
+               (find-peak list right)) ;; Recursive
+
+             (:value current-item)))))))
+
+(defn- ^:dynamic peak?
+  "determines if an item is a peak compared to the supplied comparitors"
+  [item & comparitors]
+  (let [item-value (:value item)
+        values (map (fn [item] (:value item)) comparitors)]
+    (if (every? (fn [v] (>= item-value v)) values) item-value -1)))
+
+(defn- item
+  "returns a map that represents an item in a list. The data describes the
+  index at which the item resides in the list and it's value. e.g. {index 1 :value 3}"
+  [list, index]
+  {:index index :value (nth list index nil)})
+
+(defn- right-neighbour
+  "returns the right neighbour of an item in a list"
+  [list current-item]
+  (item list (+ (:index current-item) 1)))
+
+(defn- left-neighbour
+  "returns the left neighbour of an item in a list"
+  [list current-item]
+  (item list (- (:index current-item) 1)))
+
+(defn- middle-val
+  "returns the middle value of a list"
+  [list]
+  (-> (count list) (- 1) (/ 2) (int)))
+
+
+
+;; Unit tests
+(ns peakfinder-test
+  (:use clojure.test)
+  (:use [clojure.tools.trace :only [dotrace]])
+  (:use peakfinder))
+
+(deftest finding-a-peak
+  (is (contains? (set '(7)) (find-peak '(3 2 4 7 5 6 2))))
```

Comment:
You can use the set literal syntax here.
(set '(3 5 2)) is the same as #{3 5 2}.

Also, you should almost always use vector literals rather than quoted lists when you have a literal sequence.  (of course in this case you should have used the set literal as explained above).  So instead of '(5 4 2 2) you should use [5 4 2 2].  They are both seqs and can be consumed by any of the functions in the seq library.  Vectors and Lists do have different performance characteristics which you should look up and become familiar with.  Also, the vector and list versions differ in this important way.  Each item in the vector version is evaluated and put into the vector.  The quoted list version quotes the whole thing so items are not evaluated.  In the above example that makes no difference, but here is an example where it does matter.

[(+ 1 2) (+ 2 3)] => [3 5]

While
 '((+ 1 2) (+ 2 3))  => ((+ 1 2) (+ 2 3))

The quoted list just evaluates to exactly itself.  That's what quoting does.  So the resulting list contains the two items.  The first item is a list whose contents are 1) the symbol "+"  2) the number 1 and 3) the number 2.